File: Backend/src/models/user_linked_sites.py
# Backend/src/models/user_linked_sites.py
# Definition of the UserLinkedSite model, connecting users and linked sites

# Author: Valentin Haas, 2024

# Library imports
from sqlalchemy import Column, Integer

# Project imports
from ._base import BaseTable


class UserLinkedSitesTable(BaseTable):
    """Definition of the UserLinkedSite model for the database."""

    __tablename__ = "UserLinkedSites"

    User = Column(Integer, primary_key=True, foreign_key="Users.Id")
    """The unique identifier of the user."""

    Site = Column(Integer, primary_key=True, foreign_key="LinkedSites.Id")
    """The unique identifier of the linked site."""

File: Backend/src/models/userlinkedsites.go
// Backend/src/models/userlinkedsites.go
// Definition of the UserLinkedSite model, connecting users and linked sites

// Author: Valentin Haas, 2025
package models

import (
	"gorm.io/gorm"
)

// UserLinkedSite defines the relationship between users and linked sites.
type UserLinkedSite struct {
	gorm.Model            // Provides ID, CreatedAt, UpdatedAt, DeletedAt fields
	User       User       // Referenced User Object
	UserID     uint       `gorm:"primaryKey"` // The user in the relationship (foreign key reference to User.ID).
	Site       LinkedSite // Referenced LinkedSite Object
	SiteID     uint       `gorm:"primaryKey"` // The linked site in the relationship (foreign key reference to LinkedSite.ID).
}
